src/main/java/net/consensys/orion/http/handler/privacy/AddToPrivacyGroupHandler.java

```diff
@@ -0,0 +1,164 @@
+/*
+ * Copyright 2019 ConsenSys AG.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+package net.consensys.orion.http.handler.privacy;
+
+import static net.consensys.orion.http.server.HttpContentType.JSON;
+
+import net.consensys.cava.crypto.sodium.Box;
+import net.consensys.orion.config.Config;
+import net.consensys.orion.enclave.Enclave;
+import net.consensys.orion.enclave.PrivacyGroupPayload;
+import net.consensys.orion.enclave.QueryPrivacyGroupPayload;
+import net.consensys.orion.exception.OrionErrorCode;
+import net.consensys.orion.exception.OrionException;
+import net.consensys.orion.http.handler.set.SetPrivacyGroupRequest;
+import net.consensys.orion.network.ConcurrentNetworkNodes;
+import net.consensys.orion.network.NodeHttpClientBuilder;
+import net.consensys.orion.storage.Storage;
+import net.consensys.orion.utils.Serializer;
+
+import java.util.Arrays;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.atomic.AtomicReference;
+import java.util.stream.Stream;
+
+import io.vertx.core.Handler;
+import io.vertx.core.Vertx;
+import io.vertx.core.buffer.Buffer;
+import io.vertx.ext.web.RoutingContext;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+/**
+ * Add a member to a privacy group
+ */
+public class AddToPrivacyGroupHandler extends PrivacyGroupBaseHandler implements Handler<RoutingContext> {
+
+  private static final Logger log = LogManager.getLogger();
+
+  private final Storage<PrivacyGroupPayload> privacyGroupStorage;
+  private final Storage<QueryPrivacyGroupPayload> queryPrivacyGroupStorage;
+  private final Enclave enclave;
+
+
+  public AddToPrivacyGroupHandler(
+      final Storage<PrivacyGroupPayload> privacyGroupStorage,
+      final Storage<QueryPrivacyGroupPayload> queryPrivacyGroupStorage,
+      final ConcurrentNetworkNodes networkNodes,
+      final Enclave enclave,
+      final Vertx vertx,
+      final Config config) {
+    super(networkNodes, NodeHttpClientBuilder.build(vertx, config, 1500));
+    this.privacyGroupStorage = privacyGroupStorage;
+    this.queryPrivacyGroupStorage = queryPrivacyGroupStorage;
+    this.enclave = enclave;
+  }
+
+  @Override
+  public void handle(final RoutingContext routingContext) {
```

Comment:
it would potentially help readability to split this function into smaller chunks - ideally sub functions with ~10 lines? Smaller if it's possible.

Reply:
I've split this into smaller chunks, but it could certainly benefit from further optimization. However there are quite a few other PRs and features that I'm working on so depending on feedback I'll re-prioritize this in a future PR.